hooks/wordpress/useWordpressApi.ts

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2022. Heusala Group Oy <info@heusalagroup.fi>. All rights reserved.
+
+import { WordpressService } from "../../services/WordpressService";
+import { useCallback, useEffect, useState, Dispatch, SetStateAction } from "react";
+import { WordpressPageDTO } from "../../../core/wordpress/dto/WordpressPageDTO";
+import { LogService } from "../../../core/LogService";
+import { WordpressReferenceDTO } from "../../../core/wordpress/dto/WordpressReferenceDTO";
+import { WordpressUserProfileDTO } from "../../../core/wordpress/dto/WordpressUserProfileDTO";
+
+const LOG = LogService.createLogger('useWordpressList');
+
+export function useWordpressList():
```

Comment:
You should probably create three hooks, one for each, and also have a refresh callback.

This current design would force to call all three even if you only need one.

Reply:
Functionality splitted to three hooks, and added refresh callback